src/ui/map.js

```diff
@@ -742,6 +742,54 @@ class Map extends Camera {
             this.dragRotate.isActive();
     }
 
+    _createDelegatedListener(type: MapEvent, layerId: any, listener: any) {
+        if (type === 'mouseenter' || type === 'mouseover') {
+            let mousein = false;
+            const mousemove = (e) => {
+                const features = this.getLayer(layerId) ? this.queryRenderedFeatures(e.point, {layers: [layerId]}) : [];
+                if (!features.length) {
+                    mousein = false;
+                } else if (!mousein) {
+                    mousein = true;
+                    listener.call(this, new MapMouseEvent(type, this, e.originalEvent, {features}));
+                }
+            };
+            const mouseout = () => {
+                mousein = false;
+            };
+            return {layer: layerId, listener, delegates: {mousemove, mouseout}};
```

Comment:
`_createDelegatedListener` returns `layer` and `listener`, but they're not used anywhere; instead the calling code accesses only the `delegates subobject. Let's remove these properties and return the delegates directly.

Reply:
@kkaefer it looks like those properties are used in `removeDelegatedListener`, no?  https://github.com/mapbox/mapbox-gl-js/blob/ffe7d5d0488e13ef39629b3e46416a4a4e1edd0e/src/ui/map.js#L914

Reply:
🤦‍♂ you're right